esp-config/src/generate/mod.rs

```diff
@@ -246,15 +258,93 @@ pub struct ConfigOption {
     /// Inactive options are not included in the documentation, and accessing
     /// them provides the default value.
     pub active: bool,
+
+    /// A display hint (for tooling)
+    pub display_hint: DisplayHint,
 }
 
 impl ConfigOption {
+    /// An integer config option
+    ///
+    /// Unstable and active by default.
+    pub fn integer(name: &str, description: &str, default_value: i128) -> Self {
```

Comment:
I would like to see the ConfigOption changes in a separate PR. I'm also wondering if we could get rid of the different constructors and decide based on the type of the value instead.

Reply:
I think turning the big structs into builders is, well, not that valuable but I'm also not against it and you clearly think it's a good change so I'll not argue against it (except to say that I like named fields and the `string` constructor being 3 stirngs is slightly-but-not-really harder to understand). But this is a huge PR and that part is a bit of a logical unit, so if we can split it out to shrink this one, that is worth it to me.